### ISSUE TITLE
eslint TypeScript rules overrides requires the "overrides" property to be used

### ISSUE THREAD

**Issue by ranyitz** [wix/yoshi#2362]
### 😯 Current Behavior
When trying to modify `eslint-config-yoshi`'s TypeScript rules, it needs to be done via `overrides` config instead of just using the rule regularly, this happens due to the way it was implemented.

```json
"eslintConfig": {
  "extends": "yoshi",
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {
        "dot-notation": "off"
      }
    }
  ]
}
```
See discussions in slack:

* https://wix.slack.com/archives/CAL591CDV/p1589099059483500?thread_ts=1588937683.473700&cid=CAL591CDV

* https://wix.slack.com/archives/CAL591CDV/p1587999905335300?thread_ts=1587990712.332100&cid=CAL591CDV

### 🤔 Expected Behavior

This should work:
```json
"eslintConfig": {
  "extends": "yoshi",
  "rules": {
    "dot-notation": "off"
  }
}
```
### 💁 Possible Solution

Change the way this was implemented in Yoshi (to not use overrides). The only problem is that users are aware of this (in their eslint configuration overrides) and therefore this would be a breaking change.

We can use Yoshi's Next major version to fix that.

**Comment by amitdahan** [comment]
> We can use Yoshi's Next major version to fix that.

@ranyitz We already accepted a breaking change when we decided to update `prettier`'s major.
Wdyt about this one?

**Comment by ranyitz** [comment]
@amitdahan I agree. I think that the potential damage is quite small, we can announce this before we release. PRs welcome!
